### PR TITLE
E2e financiers

### PR DESCRIPTION
**Pull request by Mussuranae**
E2E test for investment email

Related to #6210 
image

### CODE REVIEW

File: apps/financiers/financiers-e2e/src/integration/app.spec.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ describe('financiers', () => {
   });
 
   it('should display welcome message', () => {
-    cy.wait(300 * SEC);
+    cy.wait(30 * SEC);
     cy.get('financiers-landing h4', {timeout: 30 * SEC})
```
Comment (xmano) [inline apps/financiers/financiers-e2e/src/integration/app.spec.ts:13]: `cy.get('financiers-landing h4', {timeout: 30 * SEC})`  you could increate timeout here and eliminate the previous wait. This is equivalent and will be faster.

File: apps/financiers/financiers/src/app/marketplace/home/home.component.html [hunk 2]
```diff
@@ -8,8 +8,8 @@ <h1 class="mat-display-2">The professional financing hub <br />for films and TV
     professional content financiers.
   </p>
   <div class="actions" fxLayout="row" fxLayoutGap="24px">
-    <a mat-flat-button color="accent" test-id="library" routerLink="../title">Projects</a>
-    <a mat-stroked-button test-id="sales agent" routerLink="../organization">Financiers</a>
+    <a mat-flat-button color="accent" routerLink="../title">Projects</a>
+    <a mat-stroked-button routerLink="../organization">Financiers</a>
```
Comment (xmano) [inline apps/financiers/financiers/src/app/marketplace/home/home.component.html:12]: Why did you remove test-id here? Is it not needed?

File: libs/e2e/src/lib/utils/functions.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ export function clearDataAndPrepareTest(path: string = '/auth') {
   cy.clearLocalStorage();
   indexedDB.deleteDatabase('firebaseLocalStorageDb');
   if (path !== '') {
-    cy.visit(path);
+    cy.visit(path, {timeout: 30 * SEC});
```
Comment (xmano) [inline libs/e2e/src/lib/utils/functions.ts:12]: This may be counterproductive.

https://docs.cypress.io/guides/references/configuration#Timeouts

pageLoadTimeout is already set at 60s. I suggest you remove this since our servers are slower and we need 60s timeout.

File: apps/financiers/financiers-e2e/src/support/pages/marketplace/SearchPage.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+
+  ///////////////////////////
+  // ACTIONS ON MOVIE CARD //
+  ///////////////////////////
+  public clickWishlistButton(movieName: string) {
+    cy.get('movie-card', {timeout: 30 * SEC})
+      .contains(movieName).parent().parent()
+      .find('button[test-id=heart-button]').click('bottom', {force: true} );
+    cy.wait(2000);
```
Comment (xmano) [inline apps/financiers/financiers-e2e/src/support/pages/marketplace/SearchPage.ts:44]: 2 * SEC will improve readability of wait time

File: apps/financiers/financiers-e2e/src/support/pages/marketplace/SearchPage.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  /** Click on a movie-card and navigate to the movie-view */
+  public selectMovie(movieName: string) {
+    cy.log(`=>selectMovie : {${movieName}}`);
+    cy.get('movie-card', {timeout: 30 * SEC})
+      .contains(movieName).parent().parent()
+      .find('a').click();
```
Comment (xmano) [inline apps/financiers/financiers-e2e/src/support/pages/marketplace/SearchPage.ts:52]: I suggest a small wait here of 0.5 SEC after click

File: libs/e2e/src/lib/utils/functions.ts [hunk 6]
```diff
@@ -229,3 +229,11 @@ export function assertMoveTo(path: string) {
     expect(loc.pathname).to.eq(path);
   });
 }
+
+/** Used to navigate from the menu in the navbar */
+export function openSidenavMenuAndNavigate(button: string = '') {
```
Comment (xmano) [inline libs/e2e/src/lib/utils/functions.ts:234]: Can this be achieved by existing function `clickOnMenu()` ?

File: apps/financiers/financiers-e2e/src/integration/investment-email.spec.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    p4.fillDiscussionForm(discussionData);
+    p4.sendDiscussionEmail();
+
+    // Waiting this snackbar to appear, because it appears after the emails have been sent.
+    cy.get('snack-bar-container', {timeout: 30 * SEC}).should('contain', 'Your email has been sent.');
+    cy.log('Email sent');
+
+    // Check if emails are well sent.
+    cy.log('Checking emails...');
+    cy.mailosaurSearchMessages(serverId, {
```
Comment (xmano) [inline apps/financiers/financiers-e2e/src/integration/investment-email.spec.ts:60]: Would be good to check the subject line of the email and content if it is unique.

**Review by xmano — CHANGES_REQUESTED** [review]
In some place we have timeout: 20000 etc - please use 20 * SEC to represent the values as it is easier to read.

File: apps/financiers/financiers/src/app/marketplace/home/home.component.html (hunk 2 above)
Comment (Mussuranae) [reply apps/financiers/financiers/src/app/marketplace/home/home.component.html:12]: No it's not used, and that's created some duplicate with the sidenav.

File: libs/e2e/src/lib/utils/functions.ts (hunk 6 above)
Comment (Mussuranae) [reply libs/e2e/src/lib/utils/functions.ts:234]: Yes it can, so we have a duplicate here I guess, because this function is also on some catalog e2e I think.

File: apps/financiers/financiers-e2e/src/integration/investment-email.spec.ts (hunk 7 above)
Comment (Mussuranae) [reply apps/financiers/financiers-e2e/src/integration/investment-email.spec.ts:60]: There is the name of the user who send the email, I don't know if I can get it in e2e :/ 
I can do a partial check because the rest of the subject is more or less static.

File: apps/financiers/financiers-e2e/src/integration/app.spec.ts (hunk 1 above)
Comment (Mussuranae) [reply apps/financiers/financiers-e2e/src/integration/app.spec.ts:13]: I wonder if we want this test as it only goes on the landing page and look if there is the title....

Comment (xmano) [reply apps/financiers/financiers-e2e/src/integration/app.spec.ts:13]: For now let the test remain there. I remember putting the delay as workaround for CI Issue. Now that is fixed and we can remove line 12 entirely and add 60s timeout for get. 

We will need this test later for implementing sanity checks for deploy. So let it be.

**Review by xmano — APPROVED** [review]
LGTM
